### PR TITLE
maint: migrate to use github actions 

### PR DESCRIPTION
**Pull request by ccortezaguilera**
Since we don't want to maintain the repo in travis.ci and because the previous owners have moved on we are going to move to using github actions.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -4,264 +4,265 @@
...
+        "@jridgewell/gen-mapping": "^0.1.0",
+        "@jridgewell/trace-mapping": "^0.3.9"
+      }
+    },
     "@babel/code-frame": {
-      "version": "7.12.13",
-      "resolved": "https://artifactory.nerdwallet.io/artifactory/api/npm/npm/@babel/code-frame/-/code-frame-7.12.13.tgz",
-      "integrity": "sha1-3PyCa+72XnXFDiHTg319lXmN1lg=",
+      "version": "7.18.6",
+      "resolved": "https://registry.npmjs.org/@babel/code-frame/-/code-frame-7.18.6.tgz",
```
Comment (ccortezaguilera) [inline package-lock.json:19]: this needed to go to the public registry as did a lot of the packages.

Comment (nwalters512) [reply package-lock.json:19]: Aside: should we consider moving to Yarn like we did internally?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: ['12', '14', '16']
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
```
Comment (nwalters512) [inline .github/workflows/ci.yml:18]: It shouldn't be necessary to explicitly state these; they're the defaults.

```suggestion
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      - run: echo "💡 The ${{ github.repository }} repository has been cloned to the runner."
```
Comment (nwalters512) [inline .github/workflows/ci.yml:19]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+
+      - run: echo "Node v${{ matrix.node }} has been setup."
```
Comment (nwalters512) [inline .github/workflows/ci.yml:25]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - if: ${{ steps.cache-npm.outputs.cache-hit == 'false' }}
+        name: List the state of node modules
+        continue-on-error: true
+        run: npm list
```
Comment (nwalters512) [inline .github/workflows/ci.yml:46]: Is this necessary?

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
+name: Release
+on:
+  push:
+    branches:
+      - master
+jobs:
+  release:
+    if: github.event.push.repository.full_name == github.repository
```
Comment (nwalters512) [inline .github/workflows/release.yml:8]: When would we expect to see a release event for a _different_ repository?

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{github.event.push.ref}}
+          repository: ${{github.event.push.repository.full_name}}
```
Comment (nwalters512) [inline .github/workflows/release.yml:16]: These are the defaults.

```suggestion
```

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+          persist-credentials: false
```
Comment (nwalters512) [inline .github/workflows/release.yml:17]: Can you add a comment explaining why this is necessary?

File: .github/workflows/release.yml [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+      - name: Cache node modules
+        id: cache-npm
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
```
Comment (nwalters512) [inline .github/workflows/release.yml:28]: Why not use the same `Get npm cache directory` step from the `ci.yml` workflow?

File: .github/workflows/release.yml [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - if: ${{ steps.cache-npm.outputs.cache-hit == 'false' }}
+        name: List the state of node modules
+        continue-on-error: true
+        run: npm list
```
Comment (nwalters512) [inline .github/workflows/release.yml:39]: Is this necessary?

File: .github/workflows/release.yml [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Install Dependencies
+        run: npm install
+      - name: Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: npx semantic-release
```
Comment (nwalters512) [inline .github/workflows/release.yml:46]: Are you sure we want to use `semantic-release` if we're using the "Release" event to know when to publish? I'd expect us to have already bumped the version number by the time we create a new release in the GitHub UI.

File: package-lock.json (hunk 1 above)
Comment (ccortezaguilera) [reply package-lock.json:19]: if we want consistency, I think using npm here is fine.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ccortezaguilera) [reply .github/workflows/ci.yml:18]: oh really? I thought I had to specify this so that it would run for a fork PR.

Comment (nwalters512) [reply .github/workflows/ci.yml:18]: That hasn't been my experience, but maybe it's changed - can you link to the docs where you saw that?

**Comment by ccortezaguilera** [comment]
I should probably move this to a draft since it wasn't tested or ready. It was just created initially since I thought I would get access and then I didn't get access.

**Comment by ccortezaguilera** [comment]
@nwalters512 as mentioned to you, you can create your own PR and I'll simply close this so that you can work on it on your own branch. 😄

**Pull request closed without merging**